components/mediation-admin/org.wso2.carbon.message.processor/src/main/java/org/wso2/carbon/message/processor/service/MessageProcessorAdminService.java

```diff
@@ -803,26 +803,29 @@ private String getMessageAsString(MessageConsumer messageConsumer) throws AxisFa
      * Pops the message from the associated queue.
      *
      * @param processorName message processor name.
+     * @return <code>true</code> if ++ is successful, else <code>false</code>
      */
-    public void popMessage(String processorName) throws AxisFault {
+    public boolean popMessage(String processorName) throws AxisFault {
         SynapseConfiguration configuration = getSynapseConfiguration();
         MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
 
         try {
-            popMessageFromQueue(messageConsumer);
+            return popMessageFromQueue(messageConsumer);
         } catch (SynapseException e) {
             handleException(log, "Failed to pop message from the queue: ", e);
+            return false;
         } finally {
             messageConsumer.cleanup();
         }
     }
 
-    private  void popMessageFromQueue(MessageConsumer messageConsumer) throws AxisFault {
+    private  boolean popMessageFromQueue(MessageConsumer messageConsumer) throws AxisFault {
         try {
             messageConsumer.receive();
-            messageConsumer.ack();
+            return messageConsumer.ack();
         } catch (SynapseException e) {
            handleException(log, "MessageConsumer failed to receive or acknowledge the message :" , e);
+           return false;
```

Comment:
Is this statement reachable? IIRC, handle exception thorws and exception and the code will not reach here. Please correct me if I'm wrong.

Reply:
yes. Technically it shouldn't reach that particular line, but the IDE throws an error saying "Missing return statement" without this line. 

If the return line is placed before the `handleException()` , I don't think handleExeption will get executed.

Reply:
yes, you cannot have any code after a return statement in the same flow. Also, I checked without the return statement and it gives a compilation error. The workaround would be to create an exception and return it insideHandleException(and rename the mthod) rather than throwing it inside the method itself.